apps/autonolas-registry/common-util/apiRoute/service-activity.ts

```diff
@@ -0,0 +1,201 @@
+
+  return {
+    activityType: 'Demand',
+    requestId,
+    requestIpfsHash,
+    requestBlockTimestamp,
+    requestedBy,
+    requestTransactionHash,
+    deliveryIpfsHash,
+    deliveredBy,
+    deliveryTransactionHash,
+    deliveryBlockTimestamp,
+  };
+};
+
+const convertDeliveryToActivity = (delivery: Delivery): Activity => {
+  const {
+    id: requestId,
+    ipfsHash: deliveryIpfsHash,
+    mech: deliveredBy,
+    sender: requestedBy,
+    blockTimestamp: deliveryBlockTimestamp,
+    transactionHash: deliveryTransactionHash,
+    request,
+  } = delivery || {};
+  const {
+    ipfsHash: requestIpfsHash,
+    blockTimestamp: requestBlockTimestamp,
+    transactionHash: requestTransactionHash,
+  } = request || {};
+
+  return {
+    activityType: 'Supply',
+    requestId,
+    requestIpfsHash,
+    requestBlockTimestamp,
+    requestedBy,
+    requestTransactionHash,
+    deliveryIpfsHash,
+    deliveredBy,
+    deliveryTransactionHash,
+    deliveryBlockTimestamp,
+  };
+};
+
+const sortActivities = (activities: Activity[]) => {
+  return activities.sort((activityA, activityB) => {
+    // Get the relevant timestamp for each activity
+    const getTimestamp = (activity: Activity) => {
+      return activity.activityType === 'Demand'
+        ? Number(activity.requestBlockTimestamp)
+        : Number(activity.deliveryBlockTimestamp);
+    };
+
+    const timestampA = getTimestamp(activityA);
+    const timestampB = getTimestamp(activityB);
+
+    // Sort in descending order (latest timestamp first)
+    return timestampB - timestampA;
+  });
+};
+
+type ActivityResponse = {
+  id: string;
+  requests: Request[];
+  delivers: Delivery[];
+};
+
+export const mergeServiceActivity = (
+  activityFromMM: ActivityResponse,
+  activityFromLegacy: ActivityResponse,
+) => {
+  const { id } = activityFromMM || {};
+  const { requests: requestsFromMM, delivers: deliveriesFromMM } = activityFromMM || {};
+  const { requests: requestsFromLegacy, delivers: deliveriesFromLegacy } = activityFromLegacy || {};
+
+  const requestActivitiesFromMM = (requestsFromMM || [])?.map(convertRequestToActivity);
+  const requestActivitiesFromLegacy = (requestsFromLegacy || [])?.map(convertRequestToActivity);
+
+  const deliveryActivitiesFromMM = (deliveriesFromMM || [])?.map(convertDeliveryToActivity);
+  const deliveryActivitiesFromLegacy = (deliveriesFromLegacy || [])?.map(convertDeliveryToActivity);
+
+  const activities = [
+    ...requestActivitiesFromMM,
+    ...requestActivitiesFromLegacy,
+    ...deliveryActivitiesFromMM,
+    ...deliveryActivitiesFromLegacy,
+  ];
+
+  const sortedActivities = sortActivities(activities);
+
+  return {
+    id,
+    activities: sortedActivities,
+  };
+};
+
+export const getServiceActivityFromMMSubgraph = async ({
+  network,
+  serviceId,
+}: {
+  network: Network;
+  serviceId: string;
+}) => {
+  const client = network === 'gnosis' ? MM_GNOSIS_GRAPHQL_CLIENT : MM_BASE_GRAPHQL_CLIENT;
```

Comment:
could have a map and get map[network], like in website https://github.com/valory-xyz/olas-website/blob/main/common-util/graphql/client.js#L34

Reply:
otherwise if not base or gnosis, we will fallback to base, which is also a bit concerning, could lead to the same issues that on details page where we fall back to gnosis